src/Common/LRUCache.h

```diff
@@ -113,18 +149,39 @@ class LRUCache
 
         /// Insert the new value only if the token is still in present in insert_tokens.
         /// (The token may be absent because of a concurrent reset() call).
-        bool result = false;
+        bool is_value_loaded = false;
+        bool is_value_updated = false;
         auto token_it = insert_tokens.find(key);
         if (token_it != insert_tokens.end() && token_it->second.get() == token)
         {
-            setImpl(key, token->value, cache_lock);
-            result = true;
+            // setImpl() may fail, but the final behavior seems not be affected
+            // next call of getOrTrySet() will still call load_func()
+            is_value_updated = setImpl(key, token->value, cache_lock);
+            is_value_loaded = true;
```

Comment:
@alexey-milovidov I don't know what to do with these two almost identical variables. The difference is that previously if we put key in cache it could be removed just after we did that by calling `removeOverflow()` at the end of `setImpl`. And `is_value_loaded` could  mean that it was loaded but removed from cache right after. Therefore now appears another variable (`is_value_updated`) to check that we loaded and actually managed to put it.
I would rather have one variable instead of two almost identical, but this would change previous behaviour of `getOrSet` method a bit.

Reply:
Resolved, not possible to unite.